### ISSUE TITLE
March of machines when played sends every non creature artifact to the graveyard

### ISSUE THREAD

**Issue by melvinzhang** [magarena/magarena#16]
originally reported on http://www.slightlymagic.net/forum/viewtopic.php?f=82&t=3735&p=167489#p167308

muaddib wrote:
March of machines doesn't work.

jerichopumpkin wrote:
uh-oh, that's my fault, didn't check it before submit, since it was Opalescence clone (only the filter is different). I'll check it later and resubmit

Edit: can't seem to find what's wrong with it. The groovy code seems fine (exactly the same as Opalescence, except for the accept condition that is not necessary and for the filter). From my tests, it seems to work, but the converted mana cost it gets is alwys zero (as soon as I play it, every non creature artifact is sent to the graveyard).

**Comment by ShawnieBoy** [comment]
This would have to determine and compare each artfact's original type with it's current type, check that it's current creature type isn't being applied by something else, then set a power/toughness. :frowning:

**Comment by ShawnieBoy** [comment]
Ok, this is an improvement, not perfect, but better than functioning as mass artifact removal.

**Comment by PalladiaMors** [comment]
For fairness: Jericho took the blame, but I recall submitting this card as well. I think we both may have ended up submitting it more or less at the same time. I saw he did Opalescence, got excited with the strong card and changed the filter for MotM. I actually did playtest it. To me it seemed that a key difference between this and Opalescence was the fact that there are lots of 0 cost artifacts and afaik there's no 0 cost enchantment, so I wanted to ensure it was working properly for those and playtested against a deck full of Moxes. They got destroyed alright, hence I thought the card was fine. I had actually stumbled into the only case it worked properly. Anyway, I'll share the blame with Jericho.
